core/kernel/ldelf_syscalls.c

```diff
@@ -133,17 +136,21 @@ TEE_Result ldelf_syscall_open_bin(const TEE_UUID *uuid, size_t uuid_size,
 		sess->user_ctx = sys_ctx;
 	}
 
+	res = memdup_user(uuid, uuid_size, (void **)&uuid_kbuf);
+	if (res)
+		return res;
+
 	binh = calloc(1, sizeof(*binh));
```

Comment:
Is freeing `binh` missing in the error exit path?
If so, it is a bug unrelated to this commit and should be fixed seperatly (aka in a dedicated commit).

Reply:
iiuc `binh` is freed within `binh_close()`?